datafusion/physical-plan/src/sorts/merge.rs

```diff
@@ -127,12 +148,18 @@ impl<C: CursorValues> SortPreservingMergeStream<C> {
             metrics,
             aborted: false,
             cursors: (0..stream_count).map(|_| None).collect(),
+            prev_cursors: (0..stream_count).map(|_| None).collect(),
+            round_robin_tie_breaker_mode: false,
+            num_of_polled_with_same_value: vec![0; stream_count],
```

Comment:
Do we really need to keep the counts or can we simply switch to other partition when the same partition was chosen as winner twice in a row with equal values?

Reply:
I agree that storing 0 and 1 (or boolean) is enough, but I think it is less understandable and the improvement might not be worth it.

Reply:
If you are thinking about storing a single previous partition index, it doesn't work for partition numbers larger than 2.
 Say there are 3. We choose 0 and 1, ideally we should choose 2 next but unfortunately getting back to 0 since the previous chosen partition is 1.

Reply:
I agree that. It would not be trivial for more than 2 partitions, and I think it would not bring a significant improvement.

Reply:
Hm yeah...

Thinking about it one more time, what about only keeping the previous partition index that was chosen previously? We can always go to the next partition:
* previous was 1, we have to chose between 0, 1 and 2: go to 2
* precious was 2, we have to chose between 0, 1 and 2, go to 1.

This way we don't have to track / compare the counts? What would be the pros/cons for thus approach?

Reply:
I think one challenge is that you assume the next index is p+1, but it could be p+r (any number). 
Consider we have 4 partition with v=100,100,120,100.

We choose 0, 1. And the next one should be 3. But if your approach, it is not easy to find the next index with the same value. It is 1+2 in this case, and it can also be 1+any number.

Reply:
Maybe one tricky way is to find the closet index that is greater than the index 🤔